spec/booking_manager_spec.rb

```diff
@@ -0,0 +1,60 @@
+require_relative 'spec_helper.rb'
+require 'pry'
+require 'minitest/skip_dsl'
+
+
+describe 'BookingManager' do
+
+  let (:sample_booking) {
+    BookingManager.new
+  }
+  #
+  let (:fully_booked) {
+    new_booking = BookingManager.new
+    20.times do
+      new_booking.make_reservation(Date.new(2018, 1, 1), Date.new(2018, 1, 7))
+    end
+    new_booking
+  }
+
+  describe '#initialize' do
+
+    it 'must create an array of 20 room integers when created' do
+      expect(sample_booking.rooms).must_be_kind_of  Array
+      expect(sample_booking.rooms.length).must_equal 20
+      expect(sample_booking.rooms[0]).must_be_kind_of Integer
+    end
+
+  end
+
+end
+
+
+describe '#find_available_rooms' do
+  # fully_booked: hotel filled from 2018, 1, 1 til 2018 1, 7
+  # - Same dates
+  it 'must raise standard error if two reservations made for the same date' do
+    expect{fully_booked.make_reservation(Date.new(2018, 1, 1), Date.new(2018, 1, 7))}.must_raise StandardError
```

Comment:
You end up referring to the variable `fully_booked` in this line...

`fully_booked` is going to be undefined.

That's because you define `fully_booked` over on line 12, in a scope that doesn't apply to these tests at all.

These tests pass because this line ends up raising an Error indeed -- an error caused by `fully_booked.make_reservation` not existing.

Reply:
Let me know if there are questions about why `fully_booked` is out of scope and undefined.